### PR TITLE
Add uninstall pipeline support for ValueFromPipelineByPropertyName

### LINKED ISSUE #626: `Uninstall-PSResource` should take `Version` from `ValueFromPipelineByPropertyName`
### Summary of the new feature / enhancement

As a user I want to be able to pipe `Get-InstalledPSResource SomeModule` to `Uninstall-PSResource`.

Currently if you try to pipe a single version to `Unisntall-PSResource` it will try to uninstall all of them:

image

Using `Uninstall-PSResource -Name SomeModule -Version 2.0.0` works, but should be pipeable ideally.

### Proposed technical implementation details (optional)

_No response_

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Add pipeline support in `Uninstall-PSResource` for `ValueFromPipelineByPropertyName` for `-Name` and `-Version` so that only specified versions are uninstalled.  Currently all versions get uninstalled since the version property isn't passed through the pipeline to `Uninstall-PSResource`.

## PR Context

Resolves #626 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: src/code/UninstallPSResource.cs [hunk 1]
```diff
@@ -25,14 +25,14 @@ public sealed class UninstallPSResource : PSCmdlet
...
-        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = NameParameterSet)]
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, ParameterSetName = NameParameterSet)]
         [ValidateNotNullOrEmpty]
         public string[] Name { get; set; }
 
         /// <summary>
         /// Specifies the version or version range of the package to be uninstalled.
         /// </summary>
-        [Parameter(ParameterSetName = NameParameterSet)]
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = NameParameterSet)]
```
Comment (alyssa1303) [inline src/code/UninstallPSResource.cs:35]: I tested `Uninstall-PSResource` on the current master branch and it actually works with pipeline input without your change, so I'm curious if the issue for the user might not come from not setting `ValueFromPipelineByPropertyName` to `true`.

File: test/UninstallPSResource.Tests.ps1 [hunk 2]
```diff
@@ -265,12 +265,26 @@ Describe 'Test Uninstall-PSResource for Modules' {
...
+        Install-PSResource -Name $testModuleName -Version "1.0.0" -Repository $PSGalleryName -TrustRepository
+        Install-PSResource -Name $testModuleName -Version "3.0.0" -Repository $PSGalleryName -TrustRepository
+        Install-PSResource -Name $testModuleName -Version "5.0.0" -Repository $PSGalleryName -TrustRepository
+
+        $res = Get-PSResource -Name $testModuleName -Version "3.0.0"
+        $res | Should -Not -BeNullOrEmpty
+
+        Get-PSResource -Name $testModuleName -Version "3.0.0" | Uninstall-PSResource
+        $res = Get-PSResource -Name $testModuleName -Version "3.0.0"
+        $res | Should -BeNullOrEmpty
```
Comment (alyssa1303) [inline test/UninstallPSResource.Tests.ps1:286]: I think it's better if you also test that versions 1.0.0 and 5.0.0 exist to ensure that Uninstall-PSResource only uninstall 1 version.

**Comment by alerickson** [comment]
I opened this PR almost 2 months ago and it looks like this is no longer an issue.  I'm searching for the PR where this issue was fixed, but I'm going to close this PR since it's no longer needed 😃

**Comment by alerickson** [comment]
Originally resolved in: https://github.com/PowerShell/PowerShellGet/pull/593

**Pull request closed without merging**